scripts/src/javatests/org/oppia/android/scripts/maven/GenerateLicenseTextsTest.kt

```diff
@@ -0,0 +1,259 @@
+    val pbFile = tempFolder.newFile("assets/test_maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val exception = assertThrows(Exception::class) { runScript() }
+    assertThat(exception).hasMessageThat().contains(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE_FAILURE)
+  }
+
+  @Test
+  fun testScript_primaryLinkTypeNeedsIntervention_failsWithNotUpToDateException() {
+    val licenseList = listOf<License>(
+      getLicense(
+        licenseName = "license_0",
+        primaryLink = INVALID_LINK,
+        primaryLinkType = PrimaryLinkType.NEEDS_INTERVENTION
+      )
+    )
+    val dependencyList = listOf<MavenDependency>(
+      getMavenDependency(
+        artifactName = "artifact:name",
+        licenseList = licenseList
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("assets/test_maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val exception = assertThrows(Exception::class) { runScript() }
+    assertThat(exception).hasMessageThat().contains(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE_FAILURE)
+  }
+
+  @Test
+  fun testScript_validDependencyList_passes() {
+    val licenseList = listOf<License>(
+      getLicense(
+        licenseName = "license_0",
+        primaryLink = VALID_LINK,
+        primaryLinkType = PrimaryLinkType.SCRAPE_DIRECTLY
+      )
+    )
+    val dependencyList = listOf<MavenDependency>(
+      getMavenDependency(
+        artifactName = "artifact:name",
+        licenseList = licenseList
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("assets/test_maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).contains(SCRIPT_PASSED_INDICATOR)
+  }
+
+  @Test
+  fun testScript_validDependencyList_generatesResourceXmlFiles() {
+    val licenseList = listOf<License>(
+      getLicense(
+        licenseName = "license_0",
+        primaryLink = VALID_LINK,
+        primaryLinkType = PrimaryLinkType.SCRAPE_DIRECTLY
+      )
+    )
+    val dependencyList = listOf<MavenDependency>(
+      getMavenDependency(
+        artifactName = "artifact:name",
+        licenseList = licenseList
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("assets/test_maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    runScript()
+
+    assertXmlGenerated("test_third_party_dependency_names.xml")
+    assertXmlGenerated("test_third_party_dependenct_versions.xml")
+    assertXmlGenerated("test_third_party_dependency_license_texts.xml")
+    assertXmlGenerated("test_third_party_dependency_license_texts_array.xml")
+    assertXmlGenerated("test_third_party_dependency_license_names_array.xml")
+  }
+
+  private fun assertXmlGenerated(filename: String) {
```

Comment:
This doesn't seem sufficient. We should probably parse the XML to verify it's valid, and perform basic checks to make sure the correct attribute names & contents are being added. At the moment, the tests don't seem to be verifying the correctness of the generated XML.

Reply:
Done.